### DISCUSSION TITLE
Request for Notification on Closing Firefox During Downloads

### DISCUSSION THREAD

**Discussion by SSSJDanny** [aclap-dev/video-downloadhelper#1509] [Help and Q&A]
### Your question

Can you please add the option to notify users when they are about to close Firefox while downloads are in progress.

A few times now, I'd be downloading videos, and without realizing it, I closed Firefox, causing me to lose all the download progress. A prompt or warning before closing Firefox would be incredibly helpful to prevent this from happening in the future.

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"f0d8daafa5b94496"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:133.0) Gecko/20100101 Firefox/133.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "C:\\Users\\Danny\\Downloads",
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 2,
    "successfull_dl": 59,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}

**Comment by Mattmjs** [comment]
If you look at the icon in the toolbar you'll see a number on it indicating a download is in progress. That should be enough to tell you
to not close the browser until the download completed. Also if you have multiple tabs open there is a setting in Firefox to warn you before closing. On browser downloading you do get a dialog but if the companion app is downloading no warning.

![cancel dialog](https://github.com/user-attachments/assets/8c8fe2ef-3a9f-4bd5-9774-224da0e184d6)
